probe/kubernetes/controls.go

```diff
@@ -144,15 +144,19 @@ func (r *Reporter) ScaleDown(req xfer.Request, resource, namespace, id string) x
 }
 
 func (r *Reporter) registerControls() {
-	controls.Register(GetLogs, r.CapturePod(r.GetLogs))
-	controls.Register(DeletePod, r.CapturePod(r.deletePod))
-	controls.Register(ScaleUp, r.CaptureResource(r.ScaleUp))
-	controls.Register(ScaleDown, r.CaptureResource(r.ScaleDown))
+	r.handlerRegistry.Lock()
+	defer r.handlerRegistry.Unlock()
+	r.handlerRegistry.RegisterLocked(GetLogs, r.CapturePod(r.GetLogs))
+	r.handlerRegistry.RegisterLocked(DeletePod, r.CapturePod(r.deletePod))
+	r.handlerRegistry.RegisterLocked(ScaleUp, r.CaptureResource(r.ScaleUp))
+	r.handlerRegistry.RegisterLocked(ScaleDown, r.CaptureResource(r.ScaleDown))
```

Comment:
Yeah, that's ugly. Maybe I should just add a batch function like:
```go
func (r *HandlerRegistry) Batch(toRemove []string, toAdd map[string]xfer.ControlHandlerFunc)
{ ... }
```
That would probably be much better than explicit locking and unlocking.